src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java

```diff
@@ -886,13 +887,60 @@ void should_insert_first_task_and_update_reconfiguration_task(String state, Stri
                 });
     }
 
+    @Test
+    public void should_test_mark_functionality_and_ignore_already_marked() {
+        TaskResource t1 = createAndSaveTask("1000000001", UUID.randomUUID().toString(), "UNASSIGNED",
+            OffsetDateTime.now().minusDays(20), "wa-ct", "wa-jd");
+        TaskResource t2 = createAndSaveTask("1000000001", UUID.randomUUID().toString(), "UNASSIGNED",
+                                             OffsetDateTime.now().minusDays(1), "wa-ct", "wa-jd");
+        List<TaskResource> tasks = Arrays.asList(t1, t2);
+        tasks.forEach(task -> await().ignoreException(AssertionFailedError.class)
+            .pollInterval(1, SECONDS)
+            .atMost(10, SECONDS)
+            .until(
+                () -> {
+                    List<ReportableTaskResource> reportableTaskList
+                        = miReportingServiceForTest.findByReportingTaskId(task.getTaskId());
+
+                    assertFalse(reportableTaskList.isEmpty());
+                    assertEquals(1, reportableTaskList.size());
+                    assertEquals(task.getTaskId(), reportableTaskList.get(0).getTaskId());
+                    assertEquals(task.getState().getValue(), reportableTaskList.get(0).getState());
+
+                    return true;
+                }));
+
+        MIReplicaDBDao miReplicaDBDao = new MIReplicaDBDao(containerReplica.getJdbcUrl(),
+                                                           containerReplica.getUsername(),
+                                                           containerReplica.getPassword());
+        // Mark only task1
+        miReplicaDBDao.callMarkReportTasksForRefresh(null, Collections.singletonList(t1.getTaskId()), null,
+                                          null, null, OffsetDateTime.now());
+
+        List<Timestamp> taskRefreshTimestamps =
+            miReplicaDBDao.callGetReplicaTaskRequestRefreshTimes(Arrays.asList(t1.getTaskId(), t2.getTaskId()));
+        Long count = taskRefreshTimestamps.stream().map(Objects::nonNull).count();
+        Assertions.assertEquals(1, count);
+
+        // Mark filters to select both the tasks
+        miReplicaDBDao.callMarkReportTasksForRefresh(null, null, null,
+                                      null, null, OffsetDateTime.now());
+
+        taskRefreshTimestamps =
+            miReplicaDBDao.callGetReplicaTaskRequestRefreshTimes(Arrays.asList(t1.getTaskId(), t2.getTaskId()));
+        count = taskRefreshTimestamps.stream().map(Objects::nonNull).count();
+        Assertions.assertEquals(2, count);
+        // verify marked times of both the tasks are different
+        Assertions.assertNotEquals(taskRefreshTimestamps.get(0), taskRefreshTimestamps.get(1));
```

Comment:
Can using explicit requestTime allow assertion that that requestTime is clearly left alone after new refresh requset?